src/mender-update/daemon/states.hpp

```diff
@@ -58,25 +58,34 @@ class IdleState : virtual public StateType {
 
 class PollForDeploymentState : virtual public StateType {
 public:
-	PollForDeploymentState(events::EventLoop &loop) :
-		poll_timer_(loop) {
-	}
+	PollForDeploymentState(
+		events::EventLoop &event_loop, int retry_interval_seconds, int retry_count);
+
 	void OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) override;
 
 private:
-	events::Timer poll_timer_;
+	void HandlePollingError(Context &ctx, sm::EventPoster<StateEvent> &poster);
+	struct Retry {
+		http::ExponentialBackoff backoff;
+		events::Timer wait_timer;
+	};
+	Retry retry_;
```

Comment:
I believe this could be combined into a single statement declaring the `retry_` field as a `struct` without a specific type name. But YMMV.